app/src/main/java/ch/epfl/cs311/wanderwave/viewmodel/LoginScreenViewModel.kt

```diff
@@ -23,16 +31,30 @@ class LoginScreenViewModel @Inject constructor(private val spotifyController: Sp
   fun handleAuthorizationResponse(response: AuthorizationResponse) {
     when (response.type) {
       AuthorizationResponse.Type.TOKEN -> {
-        _uiState.value = UiState(hasResult = true, success = true)
+        authenticate(response.accessToken)
       }
       AuthorizationResponse.Type.ERROR -> {
         _uiState.value =
-            UiState(
+            uiState.value.copy(
                 hasResult = true, success = false, message = "Error logging in: ${response.error}")
       }
       else -> {
         _uiState.value =
-            UiState(hasResult = true, success = false, message = "User cancelled login")
+            uiState.value.copy(hasResult = true, success = false, message = "User cancelled login")
+      }
+    }
+  }
+
+  private fun authenticate(token: String) {
+    CoroutineScope(Dispatchers.IO).launch {
```

Comment:
I know I'm bothering many people with this but I asked chatGPT multiple times about it and it really seems better to have the viewmodel scope instead of normal coroutine scope: Here is chatGPT answer
Error handling: viewModelScope has a default SupervisorJob installed. This means that if one child job fails, other children are not affected and continue their execution.  
Memory leaks: Since viewModelScope is tied to the ViewModel lifecycle, it helps prevent memory leaks by cancelling all running coroutines when the ViewModel is cleared.

Reply:
#208